plugins/gc/scheduler.go

```diff
@@ -332,6 +332,12 @@ func (s *gcScheduler) run(ctx context.Context) {
 			// Pause threshold is always 0.0 < threshold <= 0.5
 			avg := float64(gcTimeSum) / float64(collections)
 			interval = time.Duration(avg/s.pauseThreshold - avg)
+			// anyway, on startup gcTimeSum might work fast and return 0
+			// on this case algorithm above turns in infinity loop which
+			// simple consume CPU on timer which fires without any interval
+			if interval < time.Second {
```

Comment:
Rather than define a lower bound for the interval, I think it would be better to define it for the `avg`. A sensible value there may be `5ms`. With the default threshold of 0.02 that would have the interval set at a minimum 245ms, but adjustable by changing the pause threshold.

Reply:
That still make it possible to turn into the same issue. If someone put threshold 1 for example.

My approach should help for not heavy loaded docker machine which runs on background, on that case 1s doesn't make any differences.

Reply:
> That still make it possible to turn into the same issue. If someone put threshold 1 for example.

.5 is the max value and .02 is the default. The `pauseThreshold` is the configurable value to modulate the interval though. Setting the threshold to 0.0 would also solve the problem in this case to keep the interval will stay at a second.

Better solution would be to update the default `pauseThreshold` along with changing the lower bound avg to have a better default interval for unloaded machines and generally spend less time checking the gc variables. We already have an event system that can reschedule in the cases when gc is expected to run following some user action and we see this commonly used now by clients.

Reply:
Agreed and updated the PR.

Thanks for explanation.